### PR TITLE
feat: add supported networks

### PR DESCRIPTION
**Pull request by HenryQW**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +21,7 @@
 		"fumadocs-ui": "13.4.10",
 		"lucide-react": "^0.446.0",
 		"next": "^14.2.15",
+		"node-fetch": "^3.3.2",
```
Comment (songkeys) [inline package.json]: we don't need node-fetch anymore. `fetch` is a built-in method since node 18

File: package.json [hunk 2]
```diff
@@ -9,7 +9,9 @@
 		"postinstall": "fumadocs-mdx",
 		"typecheck": "tsc --noEmit",
 		"lint": "biome check",
-		"lint:fix": "biome check --write"
+		"lint:fix": "biome check --write",
+		"generate-dsl-info": "node scripts/generate-dsl-info.mjs",
```
Comment (songkeys) [inline package.json]: write it like `generate:dsl-info` for conventions, so we can do `pnpm run generate:*` in ci

File: .github/workflows/generate-dsl-info.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          node-version: lts/*
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate DSL info
+        run: npm run generate-dsl-info
+
+      - name: Commit changes
```
Comment (songkeys) [inline .github/workflows/generate-dsl-info.yaml]: https://github.com/stefanzweifel/git-auto-commit-action

Using this action might be more convenient?

File: .github/workflows/generate-dsl-info.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Install dependencies
+        run: npm ci
```
Comment (songkeys) [inline .github/workflows/generate-dsl-info.yaml]: This repo uses pnpm. An example of setup: https://github.com/RSS3-Network/DSL-js-sdk/blob/f8ac11e5e64e101ac70b9694f326a8811e5a67bd/.github/workflows/release.yml#L18-L33

File: .github/workflows/generate-dsl-info.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Generate DSL info
+        run: npm run generate-dsl-info
```
Comment (songkeys) [inline .github/workflows/generate-dsl-info.yaml]: I think we can merge these generating actions like do `pnpm run generate:*` here

File: scripts/generate-dsl-info.mjs [hunk 6]
```diff
@@ -0,0 +1,174 @@
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import fetch from "node-fetch";
+
+// API endpoints
+const NETWORK_CONFIG_ENDPOINT =
+	process.env.NETWORK_CONFIG_ENDPOINT || "https://gi.rss3.dev";
```
Comment (songkeys) [inline scripts/generate-dsl-info.mjs]: I don't think we need different endpoints based on env here? Can just fetch from production until we need development stage docs in the future.

**Review by songkeys — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-dsl-info.yaml (hunk 5 above)
Comment (pseudoyu) [reply .github/workflows/generate-dsl-info.yaml]: Do i need any additional settings using `pnpm run generate:*` ?

i got an error `no matches found: generate:*` so i add a `pnpm run generate:all` script.

File: .github/workflows/generate-dsl-info.yaml (hunk 3 above)
Comment (pseudoyu) [reply .github/workflows/generate-dsl-info.yaml]: cool

File: .github/workflows/generate-dsl-info.yaml (hunk 5 above)
Comment (songkeys) [reply .github/workflows/generate-dsl-info.yaml]: Ah sorry. i think in pnpm, it's `pnpm run "/^generate:.*/" --parallel`

Comment (songkeys) [reply .github/workflows/generate-dsl-info.yaml]: also you might've forgot to commit the deletions of old workflow files

Comment (pseudoyu) [reply .github/workflows/generate-dsl-info.yaml]: ok that works! fixed now.

**Review by songkeys — APPROVED** [review]

**Comment by songkeys** [comment]
Well done!

**Pull request merged**